test/framework/src/main/java/org/elasticsearch/test/IndexSettingsModule.java

```diff
@@ -55,7 +55,10 @@ public static IndexSettings newIndexSettings(Index index, Settings indexSetting,
             .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
             .put(indexSetting)
             .build();
-        IndexMetadata metadata = IndexMetadata.builder(index.getName()).settings(build).build();
+        IndexMetadata metadata = IndexMetadata.builder(index.getName())
```

Comment:
Does this mean in tests, if someone chooses to use/enable fast refresh, we mark the created index as system due to this?

Reply:
Hi @pxsalehi , yes, in the unit tests (primarily engine unit tests I've noticed) that use this method ultimately. This is useful for a single stateless index engine test that uses fast_refresh, else it simply would fail; other unit tests are not using fast_refresh. Note it does not affect general integration tests -- there system indices would need to be created.

Reply:
I just wonder if this would lead to some unexpected behaviour when someone tries to add tests or debug why their test is failing. Is this the default behaviour in the production code, i.e., if one sets the fast refresh setting, the index is considered as a system setting? It seems not, since in that case we'd throw. 
In the places that use fast refresh setting, can't we explicitly use a system index instead? In any case, maybe a comment explaining the rationale behind this would be helpful.

Reply:
No it's not the default in production.

I believe if somebody wants to use the fast_refresh setting in a new unit test, they'd know they need a system setting, and that's what this change does here. The index metadata is created here for the convenience of the unit test, rather than supplied by the unit test itself (where it might have been a better choice to do it). Since this is an assistance method, I would leave it at that.

I am not sure a comment helps much since the code line is quite self-explanatory. If somebody uses fast_refresh, it makes it a system index metadata. I'll simply add `// using fast refresh requires a system index`.

Reply:
comments never hurt! 😄